MapEmbiggener.cs

```diff
@@ -456,26 +457,32 @@ private static void Postfix(Map __instance)
         }
         private static float timerStart;
         private static float rotTimerStart;
-        private static System.Collections.IEnumerator GameModes(Map instance)
+        private static IEnumerator GameModes(Map instance)
         {
             if (instance == null) { yield break; }
-            timerStart = Time.time;
-            rotTimerStart = Time.time;
+            MapPatchStartMatch.timerStart = Time.time;
+            MapPatchStartMatch.rotTimerStart = Time.time;
             while (instance != null && instance.enabled)
             {
-                if (instance != null && (float)Traverse.Create(instance).Field("counter").GetValue() > 2f && instance.size > 1f && (MapEmbiggener.chaosMode || (MapEmbiggener.suddenDeathMode && CountPlayersAlive() <= 2)))
+                if (instance != null && (float)instance.GetFieldValue("counter") > 2f && MapEmbiggener.zoomShrink > 0.3f && (MapEmbiggener.chaosMode || (MapEmbiggener.suddenDeathMode && CountPlayersAlive() <= 2)))
```

Comment:
Looking at the original code here, it's clear that I tried to stop Chaos Mode / Sudden Death Mode from shrinking too much, but that never worked. But now, I want to keep it that way; i.e. infinitely zooming in to _force_ the battle to end.